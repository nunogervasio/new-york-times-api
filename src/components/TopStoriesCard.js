import React from "react";
import { timeDifference } from "../utils/dateAndTime";
import { Header, Divider, Icon } from "semantic-ui-react";

const TopStoriesCard = ({ headline, summary, pub_date, divider_icon }) => {
  return (
    <React.Fragment>
      <Header as="h4">
        {headline}
        <Header.Subheader>{timeDifference(pub_date)}</Header.Subheader>
      </Header>
      <p>{summary}</p>
      <Divider horizontal={true}>
        <Icon name={divider_icon} />
      </Divider>
    </React.Fragment>
  );
};

export default TopStoriesCard;
