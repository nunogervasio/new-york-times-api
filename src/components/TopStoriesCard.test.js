import React from "react";
import { shallow } from "enzyme";
import TopStoriesCard from "./TopStoriesCard";

describe("TopStoriesCard", () => {
  const props = { headline: "", summary: "", pub_date: "", divider_icon: "" };
  const topStoriesCard = shallow(<TopStoriesCard {...props} />);

  it("renders properly", () => {
    expect(topStoriesCard).toMatchSnapshot();
  });
});
