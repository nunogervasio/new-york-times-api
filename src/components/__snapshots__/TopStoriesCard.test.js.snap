// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopStoriesCard renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopStoriesCard
    divider_icon=""
    headline=""
    pub_date=""
    summary=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          as="h4"
        >
          
          <HeaderSubheader>
            Invalid Date
          </HeaderSubheader>
        </Header>,
        <p>
          
        </p>,
        <Divider
          horizontal={true}
        >
          <Icon
            as="i"
            name=""
          />
        </Divider>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h4",
          "children": Array [
            "",
            <HeaderSubheader>
              Invalid Date
            </HeaderSubheader>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Invalid Date",
            },
            "ref": null,
            "rendered": "Invalid Date",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            as="i"
            name=""
          />,
          "horizontal": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "name": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            as="h4"
          >
            
            <HeaderSubheader>
              Invalid Date
            </HeaderSubheader>
          </Header>,
          <p>
            
          </p>,
          <Divider
            horizontal={true}
          >
            <Icon
              as="i"
              name=""
            />
          </Divider>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h4",
            "children": Array [
              "",
              <HeaderSubheader>
                Invalid Date
              </HeaderSubheader>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Invalid Date",
              },
              "ref": null,
              "rendered": "Invalid Date",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              as="i"
              name=""
            />,
            "horizontal": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "name": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
